testutil/server.go

```diff
@@ -113,7 +113,14 @@ func defaultServerConfig() *TestServerConfig {
 
 	ports, err := porter.RandomPorts(6)
 	if err != nil {
-		panic(err)
+		if _, ok := err.(*porter.PorterExistErr); ok {
+			// Fall back in the case that the testutil server is being used
+			// without porter. This should NEVER be used for Consul's own
+			// unit tests. See comments for getRandomPorts() for more details.
```

Comment:
If anyone on consul runs individual tests via go test without using make test this can still trigger, but that's probably fine. Can maybe print the error here so it can be seen/corrected.

Reply:
That's a good point, though a warning would spam other people's tests. I'm generally averse to silent fallbacks like this but this seems like it'll do the right thing for most places where you'd hit the fallback (and the `panic()` without `porter` running was pretty hard to grok the first time you saw it). Hopefully we can ditch this fairly soon.

Reply:
I added https://github.com/hashicorp/consul/commit/ad9425ca6353b8afcfebd19130a8cf768f7eac30 to catch the case where we somehow stop using `porter` for Consul. People who vendor `testutil` won't be running this one.